system_tests/logging_.py

```diff
@@ -68,6 +68,33 @@ def test_log_text(self):
         self.assertEqual(len(entries), 1)
         self.assertEqual(entries[0].payload, TEXT_PAYLOAD)
 
+    def test_log_text_w_metadata(self):
+        TEXT_PAYLOAD = 'System test: test_log_text'
+        INSERT_ID = 'INSERTID'
+        SEVERITY = 'INFO'
+        METHOD = 'POST'
+        URI = 'https://api.example.com/endpoint'
+        STATUS = '500'
+        REQUEST = {
+            'requestMethod': METHOD,
+            'requestUrl': URI,
+            'status': STATUS,
+        }
+        logger = Config.CLIENT.logger(self._logger_name())
+        self.to_delete.append(logger)
+        logger.log_text(TEXT_PAYLOAD, insert_id=INSERT_ID, severity=SEVERITY,
+                        http_request=REQUEST)
+        time.sleep(2)
```

Comment:
Rather than sleep for 2 seconds, why not set a max number of sleeps and call `logger.list_entries()` until we get one entry back?

```python
entries = []
retries = 1337
while not entries and retries > 0:
    retries -= 1
    entries, _ = logger.list_entries()
    if not entries:
        time.sleep(1)
```